cmd/net-crd-installer/utils/util.go

```
@@ -0,0 +1,156 @@
+		if err != nil {
+			return err
+		}
+
+		// Skip root directory i.e. config/crd/bases and any other directory since none should exist.
+		if d.IsDir() {
+			return nil
+		}
+
+		crd, err := GetCRDFromPath(crdpath, scheme)
+		if err != nil {
+			return err
+		}
+
+		group := crd.Spec.Group
+		// Check if the group equals "networking.fleet.azure.com".
+		if group == "networking.fleet.azure.com" {
+			crdName := crd.Name
+
+			if mode == "hub" {
+				crdsToInstall = append(crdsToInstall, *crd)
+			}
+
+			if mode == "member" {
+				// Apply member inclusion logic - only install specific CRDs
+				if !memberIncludedCRDs[crdName] {
+					klog.V(2).Infof("Excluding CRD %s from member cluster", crdName)
+					return nil
+				}
+				crdsToInstall = append(crdsToInstall, *crd)
+			}
+		}
+
+		return nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed to walk CRD directory: %w", err)
+	}
+
+	return crdsToInstall, nil
+}
+
+// GetCRDFromPath reads a CRD from the specified path and decodes it into a CustomResourceDefinition object.
+func GetCRDFromPath(crdPath string, scheme *runtime.Scheme) (*apiextensionsv1.CustomResourceDefinition, error) {
+	// Read and parse CRD file.
+	crdBytes, err := os.ReadFile(crdPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read CRD file %s: %w", crdPath, err)
+	}
+
+	// Create decoder for converting raw bytes to Go types.
+	codecFactory := serializer.NewCodecFactory(scheme)
+	decoder := codecFactory.UniversalDeserializer()
+
+	// Decode YAML into a structured CRD object.
+	obj, gvk, err := decoder.Decode(crdBytes, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode CRD from %s: %w", crdPath, err)
+	}
+
+	// Type assertion to make sure we have a CRD.
+	crd, ok := obj.(*apiextensionsv1.CustomResourceDefinition)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type from %s, expected %s but got %s", crdPath, gvk, apiextensionsv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
```

can we throw the real unexpected error?